# [\#9667 Issue](https://github.com/ethereum/solidity/issues/9667) `closed`: [SMTChecker] std::out_of_range exception because key not found in map
**Labels**: `bug :bug:`


#### <img src="https://avatars.githubusercontent.com/u/2388185?v=4" width="50">[bshastry](https://github.com/bshastry) opened issue at [2020-08-25 08:49](https://github.com/ethereum/solidity/issues/9667):

## Description

The following multi source test

```
==== Source: A.sol ====
contract A { function f() public {} }
==== Source:====
import "A.sol";
pragma experimental SMTChecker;
contract C is A {}
```

throws the said exception from here:

https://github.com/ethereum/solidity/blob/29b6c1724621e01924c55febab9a9c55807e9ad7/libsolidity/formal/CHC.cpp#L936


## Environment

- Compiler version: latest develop


## Steps to Reproduce

```
$ cp test.sol test/libsolidity/syntaxTests/multiSource/import_smt.sol
$ isoltest -t syntaxTests/multiSource/import_smt
syntaxTests/multiSource/import_smt.sol: Exception during test: map::at
```

#### <img src="https://avatars.githubusercontent.com/u/2388185?v=4" width="50">[bshastry](https://github.com/bshastry) commented at [2020-08-25 08:53](https://github.com/ethereum/solidity/issues/9667#issuecomment-679895930):

I suppose the problem is SMTChecker not being enabled on the imported source, but perhaps this should be a warning or even silently skip verification?

#### <img src="https://avatars.githubusercontent.com/u/504195?u=ce2facd14af9fd474ebff49f0d44891f56f7500f&v=4" width="50">[leonardoalt](https://github.com/leonardoalt) commented at [2020-08-25 08:59](https://github.com/ethereum/solidity/issues/9667#issuecomment-679899420):

I think it's because it only considers a single SourceUnit.


-------------------------------------------------------------------------------



[Export of Github issue for [ethereum/solidity](https://github.com/ethereum/solidity). Generated on 2024.12.15 at 06:45:24.]
